pygfx/utils/load_gltf.py

```diff
@@ -103,46 +112,112 @@ class _GLTF:
                 "The `gltflib` library is required to load gltf scene: pip install gltflib"
             )
         import gltflib
 
-        path = self._path
-        self._gltf = gltflib.GLTF.load(path, load_file_resources=True)
+        if "https://" in str(path) or "http://" in str(path):
+            if not remote_ok:
+                raise ValueError(
+                    "Loading meshes from URLs is disabled. "
+                    "Set remote_ok=True to allow loading from URLs."
+                )
+            if not find_spec("httpx"):
+                raise ImportError(
+                    "The `httpx` library is required to load meshes from URLs: pip install httpx"
+                )
+
+            import httpx
+            from io import BytesIO
+            from os import path as os_path
+            import urllib.parse
+            import mimetypes
+            import asyncio
+
+            # download
+            response = httpx.get(path, follow_redirects=True)
+            response.raise_for_status()
+
+            file_obj = BytesIO(response.content)
+
+            ext = os_path.splitext(path)[1].lower()
+            if ext == ".gltf":
+                self._gltf = gltflib.GLTF.read_gltf(file_obj, load_file_resources=False)
+
+            elif ext == ".glb":
+                self._gltf = gltflib.GLTF.read_glb(file_obj, load_file_resources=False)
+
+            downloadable_resources = [
+                res
+                for res in self._gltf.resources
+                if isinstance(res, gltflib.FileResource)
+            ]
+            if downloadable_resources:
+
+                async def download_resource(res, client: httpx.AsyncClient):
+                    res_path = urllib.parse.urljoin(path, res.uri)
+                    try:
+                        response = await client.get(res_path)
+                        response.raise_for_status()
+                        res_file = BytesIO(response.content)
+
+                        res._data = res_file.read()
+                        res._mimetype = (
+                            res._mimetype or mimetypes.guess_type(res.uri)[0]
+                        )
+                        res._loaded = True
+
+                        return res
+                    except httpx.HTTPStatusError as e:
+                        gfx.utils.logger.warning(f"download failed: {e} - {res_path}")
+                    except Exception as e:
+                        gfx.utils.logger.warning(f"download failed: {e} - {res_path}")
+
+                async def _download_resources():
+                    async with httpx.AsyncClient() as client:
+                        tasks = [
+                            download_resource(res, client)
+                            for res in downloadable_resources
+                        ]
+                        results = await asyncio.gather(*tasks)
+                    return results
+
+                asyncio.run(_download_resources())
```

Comment:
This won't work if the asyncio loop is already running ... Is this function called at "runtime"?

Reply:
> Is this function called at "runtime"?

"Uh, this situation can indeed occur. In the `glTF_viewer` example, the download is performed in a separate thread."